File: tests/Generator/resource/python/complex/array_definition_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union
from .collection_definition_type import CollectionDefinitionType


# Represents an array which contains a dynamic list of values of the same type
class ArrayDefinitionType(CollectionDefinitionType):
    pass

File: tests/Generator/resource/rust/complex/array_definition_type.rs
use serde::{Serialize, Deserialize};
use collection_definition_type::CollectionDefinitionType;
use property_type::PropertyType;

// Represents an array which contains a dynamic list of values of the same type
#[derive(Serialize, Deserialize)]
pub struct ArrayDefinitionType {
    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "schema")]
    schema: Option<PropertyType>,

}

